### PR TITLE
Bookmark counter spec

### PR DESCRIPTION
**Pull request by mamrey**
A flaky test from #3215 was removed in  #3245. 

This PR fixes and replaces said spec, testing the ability to add/remove bookmarks without the bookmark navbar element present. 

These changes also add an `after` hook to clean up modifications to `CatalogController` in the bookmark icon example

Ran the bookmark test suite 100 times, all green.

### CODE REVIEW

File: spec/features/bookmarks_spec.rb [hunk 1]
```diff
@@ -96,4 +107,26 @@
...
+
+    after do
+      CatalogController.blacklight_config.navbar.partials = default_nav_partials
+    end
+
+    it 'adds and removes bookmarks', :js do
+      visit solr_document_path('2007020969')
+      expect(page).to have_no_css('#bookmarks_nav')
+      check 'Bookmark'
+
```
Comment (jcoyne) [inline spec/features/bookmarks_spec.rb:126]: Can you add a check here before navigating away from the page that shows it has been saved? Typically you can look for `In Bookmarks` in the page content.

**Review by jcoyne — APPROVED** [review]

File: spec/features/bookmarks_spec.rb (hunk 1 above)
Comment (mamrey) [reply spec/features/bookmarks_spec.rb:127]: Thanks, good call.

 I have added that check, but it fails the rails 6 check. I built an internal test app using rails 6.1.7.8 and encountered the same failure. When inspecting the app, it looks like the  `BookmarkToggle` event handler callback is fired twice in succession, so the checkbox checks and unchecks itself with one click. I haven't pinpointed the issue, but perhaps it's related to how the js assets are built/handled when using rails 6?

Comment (mamrey) [reply spec/features/bookmarks_spec.rb:127]: Some additional context: 

In the rails `6.1.7.8` test app, the browser receives both a `blacklight.source.js` and `blacklight.debug.js` file that appear the same. The `clicked` function is called in both these js files according to the debugger. This has the effect of the checkbox being clicked twice in succession, which reflects the failing test.  While I am unfamiliar with the asset pipeline for rails 6 and BL, I will investigate [turning off debug mode](https://guides.rubyonrails.org/v4.0.0/asset_pipeline.html#turning-debugging-off).

**Comment by sandbergja** [comment]
@mamrey now that main no longer supports rails 6.1 (as of #3224), would you mind rebasing this PR?  Thank you!

**Comment by mamrey** [comment]
Thank you for further investigating the issue and moving things forward @sandbergja!

**Pull request merged**